### DISCUSSION TITLE
Default Style

### DISCUSSION THREAD

**Discussion by brent-charlton** [tailwindlabs/tailwindcss#11533] [Help]
I'm trying to edit the default style - I want the default className of all my buttons to include these base classes  

className="px-3 py-1.5 rounded-full"

so that when I throw in a <button> tag its already prestyled. I've had a read through the docs but can't quite work out how to do it. When i try to @apply in my global css to the base layer. It doens't seem to work. How have others acheived this? Is it possible?

Edit: I'll add to this in that the documentation seems to advocate for repetition of the classes anyway so perhaps i've gone down a rabbit hole and should just repeat the className everywhere anyway.

**Comment by wongjn** [comment]
It should be possible. Consider providing a [Tailwind Play](https://play.tailwindcss.com) that reproduces your method that does not work.

> How have others acheived this?

Personally, I try not to touch any CSS at all when working with Tailwind. The advantage for Tailwind for me is that it allows me less context switching between the markup-generating templates and its CSS. So along with this, I would normally have some base *Button* component that has these "base" classes (but overridable).

**Reply by brent-charlton** [reply]
I thought about just creating a component and including the classes as default but it seems to be that there is likely a way to just edit the base components, so that i can in effect make a component library based off of tailwind class to save me generating a bunch of needless components for each tag i want to change.

Perhaps just a link to the correct documentation is all i need or an example where someone else has done this.

**Comment by mybearworld** [comment]
`@apply` does work ([Play](https://play.tailwindcss.com/yeR7s4JRw0?file=css)), but in general, this is not the Tailwind way to do it. I'd  suggest creating a button component with these classes applied.
